src/fields/hooks/afterRead/relationshipPopulationPromise.ts

```diff
@@ -55,6 +56,19 @@ const populate = async ({
     if (!relationshipValue) {
       // ids are visible regardless of access controls
       relationshipValue = id;
+    } else if (field.select && req.payloadAPI !== 'graphQL') {
+      const fieldsOrTrue = Array.isArray(field.select)
+        ? field.select
+        : field.select({
+          data: relationshipValue,
+          collection: relatedCollection.config,
+          siblingData: dataToUpdate,
+          req,
+        });
+      if (fieldsOrTrue !== true) {
+        // if fieldsOrTrue is true then we want to return the entire related document
+        relationshipValue = deepPick(relationshipValue, ['id', ...fieldsOrTrue]);
+      }
```

Comment:
Can't comment on unchanged lines in Github, but line 118 should be changed to:

`if (req.locale === 'all' && typeof siblingDoc[field.name] === 'object' && siblingDoc[field.name] !== null) {`

since typeof 'object' will be true with null, leading to the Object.keys[] to fail in the next line.

Reply:
Yep. One of those awesome JS things, it hurts you once and you never forget.

Reply:
This was a different error and was already fixed